src/main/peripherals/Lock.java

```diff
@@ -0,0 +1,46 @@
+package peripherals;
+import java.io.IOException;
+import java.net.InetAddress;
+
+
+
+public class Lock implements IPeripheral{	
+	private String ip;
+	private String port;
+	
+	public Lock(String ip, String port){
+		this.ip = ip;
+		this.port = port;
+	}
+	
+
+	public Lock(InetAddress ip, int port){
+		this.ip = ip.toString();
+		this.port = Integer.toString(port);
+	}
+	
+	@Override
+	public void performAction(String action) throws ActionNotDefinedException, IOException, InterruptedException{
+		if(action.equals("openLock")){
```

Comment:
The code found in this class is specific for this type of peripheral and will not be reused for any other peripherals. For a different peripherals, for instance a philips hue light we would create a new implementation of the IPeripheral class with methods regarding the different actions the light could perform, these methods are then called using a single string that is passed into the performAction method.